src/Swarm/App.hs

```diff
@@ -18,6 +18,7 @@ import           Control.Concurrent   (forkIO, threadDelay)
 
 import           Brick
 import           Brick.BChan
+import           Brick.Main
```

Comment:
This new import seems redundant, could you remove it?

Reply:
Could we add -Wall to the library ghc-options to show those error?

Reply:
Yes, that's probably a good idea.  We might even consider adding `-Wall -Werror` to the CI.  I really like trying to keep things `-Wall`-clean, I have found many bugs over the years by forcing myself to pay attention to warnings.